community/kernel-api/src/test/java/org/neo4j/internal/kernel/api/NodeValueIndexCursorTestBase.java

```diff
@@ -166,6 +183,27 @@ public void shouldPerformExactLookup() throws Exception
         }
     }
 
+    @Test
+    public void shouldPerformExactLookupInCompositeIndex() throws Exception
+    {
+        // given
+        int label = token.nodeLabel( "Person" );
+        int firstName = token.propertyKey( "firstname" );
+        int surname = token.propertyKey( "surname" );
+        CapableIndexReference index = schemaRead.index( label, firstName, surname );
+        try ( NodeValueIndexCursor node = cursors.allocateNodeValueIndexCursor();
+              PrimitiveLongSet uniqueIds = Primitive.longSet() )
+        {
+            // when
+            IndexValueCapability valueCapability = index.valueCapability( ValueGroup.TEXT );
+            read.nodeIndexSeek( index, node, IndexOrder.NONE, IndexQuery.exact( firstName, "Joe" ),
+                    IndexQuery.exact( firstName, "Dalton" ) );
```

Comment:
`firstName` -> `surname`... it feels unsettling that this does not cause an assertion error somewhere.